swap/src/io.rs

```diff
@@ -1,3 +1,4 @@
+// TODO: Looks like our old facade, let's be sure we learn from comit-rs
```

Comment:
What issues are we likely to run into with this Facade like struct?

Reply:
I think it's fine.
suggestions:
- rename to `Facade`
- Do not create a `Facade` trait